node/network/gossip-support/src/metrics.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use polkadot_node_subsystem_util::{
+	metrics,
+	metrics::{
+		prometheus,
+		prometheus::{Gauge, PrometheusError, Registry, U64},
+	},
+};
+
+/// Dispute Distribution metrics.
+#[derive(Clone, Default)]
+pub struct Metrics(Option<MetricsInner>);
+
+#[derive(Clone)]
+struct MetricsInner {
+	/// Tracks authority status for producing relay chain blocks.
+	is_authority: Gauge<U64>,
+	/// Tracks authority status for parachain approval checking.
+	is_parachain_validator: Gauge<U64>,
```

Comment:
It's possible to have an inconsistent state with `is_authority: 0, is_parachain_validator: 1`. But I don't know a better way of expressing an enum with prometheus (could be one metrics with 3 values, but less readable).

Reply:
The flow prevents that, because if we do not set` is_authority` to 1 we no longer check if it's a parachain validator.

Reply:
the impl was wrong, as if we are no longer authority it wouldn't uncheck the parachain validator status. Fixed it.